internal/user/repository.go

```diff
@@ -97,3 +100,53 @@ func (r *Repository) GetUserTypeByEmail(email string) (string, error) {
 	}
 	return userType, nil
 }
+
+// CreateUserWithEmail creates a new user with given email and user_type
+// returns the user struct, and an error
+func (r *Repository) CreateUserWithEmail(email, user_type string) (User, error) {
+	u := User{}
+
+	userID, err := ksuid.NewRandom()
+	if err != nil {
+		return u, err
+	}
+
+	u.ID = userID.String()
+	u.Email = email
+	u.CreatedAt = time.Now()
+	u.Type = user_type
+	u.CreatedAtHumanised = humanize.Time(u.CreatedAt.UTC())
+	if _, err := r.db.Exec(`INSERT INTO users (id, email, created_at, user_type) VALUES ($1, $2, $3, $4)`, u.ID, u.Email, u.CreatedAt, u.Type); err != nil {
+		return User{}, err
+	}
+
+	return u, nil
+}
+
+// GetUserTypeByEmailOrCreateUserIfRecruiter seeks a user in the users table or creates one and recruiter profile if the user has a job posting
+// returns user struct and an error
+func (r *Repository) GetUserTypeByEmailOrCreateUserIfRecruiter(email string, jobRepo *job.Repository, recRepo *recruiter.Repository, svr server.Server) (string, error) {
+	u, err := r.GetUserTypeByEmail(email)
+	if err == nil {
+		return u, nil
+	}
+
+	jobsCountByEmail, err := jobRepo.JobsCountByEmail(email)
+	if err != nil {
+		return "", err
+	}
+	if jobsCountByEmail > 0 {
+		user, err := r.CreateUserWithEmail(email, "recruiter")
+		if err == nil {
+			err = recRepo.CreateRecruiterProfileBasedOnLastJobPosted(user.Email, jobRepo)
+			if err != nil {
+				svr.Log(err, "unable to create recruiter profile")
```

Comment:
if we cant create the user we error and the user should not be able to signup. please remove log and return error
```suggestion
				return "", err
```

Reply:
There is a broader issue here that concerns transactions.
I chose to log it because I didn't want to fail a signon
because we could not create a recruiter's profile.

At the point I'm logging the error the user would've already been created.

The right thing to do is to perhaps run execute both the user creation 
and the profile creation in a transaction so that they both succeed together or fail together


I'll effect the changes though. Just thinking out loud here.

Reply:
fine to leave without transaction for now - if something goes wrong i will clean it up manually